src/OpenLoco/src/Paint/PaintTrack.cpp

```diff
@@ -26,6 +26,91 @@ namespace OpenLoco::Paint
     static loco_global<uint8_t, 0x00113605E> _trackTunnel;
     static loco_global<uint8_t, 0x00522095> _byte_522095;
 
+    constexpr std::array<World::Pos3, 4> kStraightBoundingBoxOffsets = {
+        World::Pos3{ 2, 5, 0 },
+        World::Pos3{ 5, 2, 0 },
+        World::Pos3{ 2, 5, 0 },
+        World::Pos3{ 5, 2, 0 },
+    };
+    constexpr std::array<World::Pos3, 4> kStraightBoundingBoxSizes = {
+        World::Pos3{ 28, 22, 1 },
+        World::Pos3{ 22, 28, 1 },
+        World::Pos3{ 28, 22, 1 },
+        World::Pos3{ 22, 28, 1 },
+    };
+    constexpr std::array<std::array<uint32_t, 3>, 4> kStraightImageIndexOffsets = {
+        std::array<uint32_t, 3>{ TrackObj::ImageIds::Style0::straightBallastNE, TrackObj::ImageIds::Style0::straightSleeperNE, TrackObj::ImageIds::Style0::straightRailNE },
+        std::array<uint32_t, 3>{ TrackObj::ImageIds::Style0::straightBallastSW, TrackObj::ImageIds::Style0::straightSleeperSW, TrackObj::ImageIds::Style0::straightRailSW },
+        std::array<uint32_t, 3>{ TrackObj::ImageIds::Style0::straightBallastNE, TrackObj::ImageIds::Style0::straightSleeperNE, TrackObj::ImageIds::Style0::straightRailNE },
+        std::array<uint32_t, 3>{ TrackObj::ImageIds::Style0::straightBallastSW, TrackObj::ImageIds::Style0::straightSleeperSW, TrackObj::ImageIds::Style0::straightRailSW },
+    };
+    constexpr std::array<uint8_t, 4> kStraightBridgeEdges = {
+        0b0101,
+        0b1010,
+        0b0101,
+        0b1010,
+    };
+    constexpr uint8_t kStraightBridgeQuarters = 0xF;
+    constexpr std::array<uint8_t, 4> kStraightTunnelsEdges = {
+        0b0101,
+        0b1010,
+        0b0101,
+        0b1010,
+    };
+    constexpr std::array<uint16_t, 4> kStraightSegments = {
+        0x0D0,
+        0x130,
+        0x0D0,
+        0x130,
+    };
+
+    // 0x004125DD & 0x0041270E
+    static void paintTrackStraight(PaintSession& session, const World::TrackElement& elTrack, const uint8_t rotation)
+    {
+        const auto height = elTrack.baseHeight();
+        const auto heightOffset = World::Pos3{ 0,
+                                               0,
+                                               height };
+        if (elTrack.hasBridge())
+        {
+            auto newBridgeEntry = BridgeEntry(
+                height,
+                0, // straight
+                kStraightBridgeEdges[rotation],
+                kStraightBridgeQuarters,
+                elTrack.bridge(),
+                ImageId::fromUInt32(_trackImageId2));
+            //newBridgeEntry.edgesQuarters |= session.getBridgeEntry().edgesQuarters;
```

Comment:
vanilla did this but i found its not required as paintTileElementsEndLoop will always reset edgesQuarters to 0 after each individual tile. I can't think of a situation where this would make a difference.

Reply:
Lol no. This is indeed required. "_paintTileElementsEndLoop will always reset edgesQuarters to 0 after each individual tile_" isn't correcct it resets it if on a different height if its the same height (i.e. a merge) it is not reset to 0.